dask/dataframe/io/parquet/arrow.py

```diff
@@ -698,10 +300,476 @@ def read_metadata(
+        )
+        if len(paths) == 1 and fs.isdir(paths[0]):
+
+            # Use _analyze_paths to avoid relative-path
+            # problems (see GH#5608)
+            base, fns = _analyze_paths(paths, fs)
+            paths = fs.sep.join([base, fns[0]])
+
+            meta_path = fs.sep.join([paths, "_metadata"])
+            if fs.exists(meta_path):
+                # Use _metadata file
+                ds = pa_ds.parquet_dataset(
+                    meta_path,
+                    filesystem=fs,
+                    partitioning=dataset_kwargs.get(
+                        "partitioning", default_partitioning
+                    ),
+                )
+                if gather_statistics is None:
+                    gather_statistics = True
+        elif len(paths) > 1:
+            base, fns = _analyze_paths(paths, fs)
+            meta_path = fs.sep.join([base, "_metadata"])
+            if "_metadata" in fns:
+                # Pyarrow cannot handle "_metadata" when `paths` is a list
+                # Use _metadata file
+                ds = pa_ds.parquet_dataset(
+                    meta_path,
+                    filesystem=fs,
+                    partitioning=dataset_kwargs.get(
+                        "partitioning", default_partitioning
+                    ),
+                )
+                if gather_statistics is None:
+                    gather_statistics = True
+
+                # Populate valid_paths, since the original path list
+                # must be used to filter the _metadata-based dataset
+                fns.remove("_metadata")
+                valid_paths = fns
+
+        if ds is None:
+            ds = pa_ds.dataset(
+                paths,
+                filesystem=fs,
+                format="parquet",
+                partitioning=dataset_kwargs.get(
+                    "partitioning", default_partitioning
+                ),  # Assume "hive" by default
+            )
+
+        if split_row_groups is None and gather_statistics is not False:
+            split_row_groups = True
+
+        # Generate list of row-group fragments and call it `metadata`
+        metadata, partition_info = _collect_pyarrow_dataset_frags(
+            ds, filters, valid_paths, fs, split_row_groups, gather_statistics
+        )
+        schema = ds.schema
+        base = ""
+
+        if not (
+            split_row_groups or gather_statistics or partition_info["partition_names"]
+        ):
+            # Don't need to process real metadata if
+            # we are not gathering statistics, splitting
+            # by row-group, or dealing with partitioned data
+            metadata = [frag.path for frag in metadata]
+
+        return (
+            schema,
+            metadata,
+            base,
+            partition_info,
+            split_row_groups,
+            gather_statistics,
+        )
+
+    @classmethod
+    def _generate_dd_meta(cls, schema, index, categories, partition_info):
```

Comment:
Can you indicate this is used by both dataset and legacy class? (and maybe for the other mehods indicating whether it is overridden below for the legacy class or not could be helpful)

Reply:
Sounds good!